### ISSUE TITLE
Let's improve the state of the art once again! 

### ISSUE THREAD

**Issue by LifeIsStrange** [getalp/disambiguate#2]
Hi, I found your paper really interesting and I would like to thank you for improving the state of the art.

But I see you used BERT which is great but is no longer the state of the art, it has been obscoleted by a significant margin by XLnet!
https://github.com/zihangdai/xlnet/

You could improve even more your WSD accuracy by reusing your same architecture but using XLnet instead of BERT. 
BTW, I believe that huggingface transformers have XLnet bindings too :)

**Comment by loic-vial** [comment]
Hi ! Thank you for your interest ! :)

We did some experiments with XLNet and RoBERTa, using the pytorch-transformers library of huggingface, but we did not see any improvement compared to BERT :( 

However, I can see at least two factors for why it did not work:

First, and I think that is the main problems, the tokenization process followed by XLNet/RoBERTa is different from BERT.
-  in BERT, they first tokenizes words and punctuations with a simple rule-based tokenizer (like the tokenizer from Moses), and then they use WordPiece to tokenize words into subwords.
- in XLNet/RoBERTa, they keep punctuations and spaces as-is, and use SentencePiece to tokenize the whole sentence into subwords.
In WSD now, if you look at the training corpora (SemCor or WNGC for instance), the punctuations are always splitted from the words. It means that if we want to have the best performance with XLNet/RoBERTa, we should detokenize the corpora and re-assign word-senses. 

Second, the hyperparameters that we chose are probably not optimal (number of Transformer layers, size of hidden layer, dropout, etc.), there is a lot to explore here :) 

I will try to commit/push our version of the code that allows to use any of huggingface's model soon, so anyone can explore all this by him/herself !

**Comment by loic-vial** [comment]
Hi again :)

We finally added support to the transformers library so anyone can use any of their provided models by using the "auto-model" and "auto-path" option ! :) 

As I said, we still did not find a model and a set of parameters that outperforms our results, but many settings remains unexplored ^^

I close the issue now but feel free to reopen it or open a new one if you have any question :)
